### DISCUSSION TITLE
3 basic problem about skyworking in python and go

### DISCUSSION THREAD

**Discussion by gujincun** [apache/skywalking#11277] [Q&A]
about skyworking-python 1.0.1 with tornado，some external request cannot show in ui ，others can i don't know why？and another problem is when i open collect mongodb param，every mongodb operation may exception?

about skyworking-go with gin，i can not enable http collect param even if i set the environment or config，why？

**Comment by Superskyyy** [comment]
Hi, thanks for raising this. As you may know agent plugins are contributed by many developers and they are may encounter bugs as the instrumented framework evolves. 

Please include details and a minimally reproducible example on what's not showing and I will fix them.

**Reply by gujincun** [reply]
about python:
python 3.8.10
apache-skyworking: 1.0.1
requests:2.31.0
tornado:6.1.0
gevent:20.12.1
pymongo:3.11.0
the exception is sw_pymongo.py line 156 in _sw_send_message
ATTribute error, '_GetMore' object has no attribe 'spec'

and i found another problem: some request should query mongodb, but no mongodb operation show in ui, why?

about skyworking-go:
go:1.20.5
gin:1.9.1
skyworking-go:0.2.0

just basic config, no other modify
 trace

**Reply by wu-sheng** [reply]
Unless you provided reproducible codes, the help would be limited. All plugins are tested only with given cases, this is how pymongo tested, https://github.com/apache/skywalking-python/tree/master/tests/plugin/data/sw_pymongo

**Reply by gujincun** [reply]
sorry, due to our it security， i cannot paste code. and i have update comment, the exception comes now.

**Reply by wu-sheng** [reply]
`reproducible codes` is not your private code. It is a new codebase to reproduce the scenarios.
There is not fun guessing what is happening in your environment. You at least should take some time. otherwise, why we should waste time.

**Reply by gujincun** [reply]
about python:
python 3.8.10
apache-skyworking: 1.0.1
requests:2.31.0
tornado:6.1.0
gevent:20.12.1
pymongo:3.11.0
the exception is sw_pymongo.py line 156 in _sw_send_message
Attribute error, '_GetMore' object has no attribe 'spec', please fouce on this exception.

some request should query mongodb, but no mongodb operation show in ui
some request should inclue external request, but not show in ui
are these Known issues？ @Superskyyy 

and my code is:
from gevent import monkey;monkey.patch_all()
from skywalking import agent, config
config.init(collector='ip:11800')
agent.start()

http_server = tornado.httpserver.HTTPServer(app)
  http_server.bind(8000)
  http_server.start(3)
  # tornado.ioloop.IOLoop.current().start()
  tornado.ioloop.IOLoop.instance().start()  # 开始事件

I see, the two requests I said went wrong, one can't show outsourced http requests, and the other can't show internal mongoDB operations, are both running on Tornado run_on_excutor, help see if there is any impact?

**Comment by Superskyyy** [comment]
Regarding go agent, FYI @mrproliu

**Comment by mrproliu** [comment]
For the Go Agent, the gin server does not support collecting HTTP parameters. However, the native `HTTP` server does support it. Could you please provide more details about your environment, including the versions of Go and SkyWalking-Go that you are using, as well as any specific environment settings?
Additionally, I would like to mention that we already have a successful test for collecting HTTP data using our [HTTP plugin](https://github.com/apache/skywalking-go/tree/main/test/plugins/scenarios/http).

**Reply by gujincun** [reply]
the versions you can see in my latest commnet, gin is famours, why cannot support collect params? we are build mircoservice with bin, so cannot instead use go sdk http, 3ks.

**Reply by wu-sheng** [reply]
> why cannot support collect params?

This is totally a culture misunderstanding. The open-source community has no obligation to do anything. Be polite! We don't owe you anything.
As I mentioned here https://github.com/apache/skywalking/discussions/11277#discussioncomment-6858831, if you don't want to spend time on this, why should we do?

**Reply by gujincun** [reply]
i dont agree. 
about skyworking-go，only add this "import _ "github.com/apache/skywalking-go"" to my code， then build with -buildtool，then run with env SW_AGENT_PLUGIN_CONFIG_HTTP_SERVER_COLLECT_PARAMETERS. 
as mrproliu, gin does not support this env.
i think you should explain in readme about features， i only see suport gin。 @mrproliu

**Reply by wu-sheng** [reply]
Funny, you are keeping asking things. But you don't ever help the community.
The author told you something, if you don't believe, why asked?

gin plugin doesn't support this, simple as this. Nothing more

If you want to contribute features, we are welcome. Don't imagine things. That variable is for http plugin, so called that.

**Reply by gujincun** [reply]
I'm not convinced, I'm saying to make it clear in the README, such as supporting the gin framework, but not supporting gin fetching http parameters... I can figure out how to implement this feature myself, and if you have the relevant information to provide better, thank you
